string2pdb/stringscore

```diff
@@ -175,19 +341,87 @@ def main():
+        interactors.to_csv(output_file, index=False, header=['Target_protein','Target_Uniprot_AC', 'StringID_Target', 'Interactor', 'Interactor_UniProt_AC', 'StringID_Interactor',
                                                          'String_score', 'Experimental_score', 'Database_score', 
-                                                         'Textmining_score'])
+                                                         'Textmining_score', 'PDB_ID','Experiment_Type','Resolution'])
+        print(f"Results saved to {output_file}")
+
+    else:
+        # Add PDB details to interactors:
+        pdb_columns = []
+        for _, interactor_row in interactors.iterrows():
+            interactor_name = interactor_row['Interactor_UniProt_AC']
+            common_pdb_ids = find_common_pdbs(target_entries, interactor_name)
+
+            if common_pdb_ids:
+                pdb_details = get_experiment_details(common_pdb_ids)
+                for pdb_info_detail in pdb_details:
+                    pdb_columns.append({
+                        'Target_protein': interactor_row['preferredName_A'],
+                        'Target_Uniprot_AC': interactor_row['Target_Uniprot_AC'],
+                        'StringID_Target': interactor_row['stringId_A'],
+                        'Interactor': interactor_row['preferredName_B'],
+                        'Interactor_UniProt_AC': interactor_row['Interactor_UniProt_AC'],
+                        'StringID_Interactor': interactor_row['stringId_B'],
+                        'String_score': interactor_row['score'],
+                        'Experimental_score': interactor_row['escore'],
+                        'Database_score': interactor_row['dscore'],
+                        'Textmining_score': interactor_row['tscore'],
+                        'PDB_ID': pdb_info_detail['PDB_ID'],
+                        'Experiment_Type': pdb_info_detail['Experiment_Type'],
+                        'Resolution': pdb_info_detail['Resolution']
+                    })
+            else:
+                pdb_columns.append({
+                    'Target_protein': interactor_row['preferredName_A'],
+                    'Target_Uniprot_AC': interactor_row['Target_Uniprot_AC'],
+                    'StringID_Target': interactor_row['stringId_A'],
+                    'Interactor': interactor_row['preferredName_B'],
+                    'Interactor_UniProt_AC': interactor_row['Interactor_UniProt_AC'],
+                    'StringID_Interactor': interactor_row['stringId_B'],
+                    'String_score': interactor_row['score'],
+                    'Experimental_score': interactor_row['escore'],
+                    'Database_score': interactor_row['dscore'],
+                    'Textmining_score': interactor_row['tscore'],
+                    'PDB_ID': None,
+                    'Experiment_Type': None,
+                    'Resolution': None
+                })
```

Comment:
many of these dictionary keys are common between the if and else dicts; can we use do them once and customise the others, so it's. bit more concise/readable? e.g.
```
this_interactor = {'Target_protein': interactor_row['preferredName_A'],
                        'Target_Uniprot_AC': interactor_row['Target_Uniprot_AC'],
                        'StringID_Target': interactor_row['stringId_A'],
                     ...}
                if common_pdb_ids:
                    pdb_details = get_experiment_details(common_pdb_ids)
                    for pdb_info_detail in pdb_details:
                        this_pdb = this_interactor.copy()
                        this_pdb['PDB_ID'] = pdb_info_detail['PDB_ID']
                        ...
else:
    this_interactor['PDB_ID'] = ...
```